### ISSUE TITLE
Not correct result from tf.split()

### ISSUE THREAD

**Issue by FGMphys** [tensorflow/tensorflow#61137]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

tf 2.8-2.9-2.10

### Custom code

No

### OS platform and distribution

Both on Mac and on Ubuntu

### Python version

3.9

### Current behavior?

Tryng to split a four axes tensor, I got some unexpected zeros in the splitted tensors.
This behaviour is not presented in the tf 2.12. Is an old bug?

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

vec=tf.ones((1,768,3,1400))
splitted=tf.split(vec,[256,512],axis=1)
print(splitted) ##zeros have occured in the tensor
```

### Relevant log output

```shell
No
```

**Comment by SuryanarayanaY** [comment]
Hi @FGMphys ,

Thanks for reaching us. I have tested the code and yes it seems bug present in TF2.10 and it got resolved in TF2.11 and later versions and attached [gist](https://colab.research.google.com/gist/SuryanarayanaY/6d55357a00c452054a54b2a975816967/61137.ipynb) for reference.. As the error resolved since last 3 versions it is unlikely to cherry pick to old versions.

Shall we mark it as closed as it resolved with latest versions. Thanks!
